wagtail_localize/views/submit_translations.py

```diff
@@ -0,0 +1,212 @@
+        context = super().get_context_data(**kwargs)
+        context.update({
+            "form": self.get_form(),
+            "next_url": self.get_success_url(),
+            "back_url": self.get_success_url() or self.get_default_success_url(),
+        })
+        return context
+
+    def post(self, request, **kwargs):
+        form = self.get_form()
+
+        if form.is_valid():
+            with transaction.atomic():
+                translator = TranslationCreator(self.request.user, form.cleaned_data["locales"])
+                translator.create_translations(self.object)
+
+                # Now add the sub tree (if the obj is a page)
+                if isinstance(self.object, Page):
+                    if form.cleaned_data["include_subtree"]:
+                        def _walk(current_page):
+                            for child_page in current_page.get_children():
+                                translator.create_translations(child_page)
+
+                                if child_page.numchild:
+                                    _walk(child_page)
+
+                        _walk(self.object)
+
+                if len(form.cleaned_data["locales"]) == 1:
+                    locales = form.cleaned_data["locales"][0].get_display_name()
+
+                else:
+                    # Note: always plural
+                    locales = _('{} locales').format(len(form.cleaned_data["locales"]))
+
+                # TODO: Button that links to page in translations report when we have it
+                messages.success(
+                    self.request, self.get_success_message(locales)
+                )
+
+                return redirect(self.get_success_url() or self.get_default_success_url())
+
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.has_perms(['wagtail_localize.submit_translation']):
+            raise PermissionDenied
+
+        self.object = self.get_object()
+        return super().dispatch(request, *args, **kwargs)
+
+
+class SubmitPageTranslationView(SubmitTranslationView):
+    title = __("Translate page")
+
+    def get_subtitle(self):
+        return self.object.get_admin_display_title()
+
+    def get_object(self):
+        return get_object_or_404(Page, id=self.kwargs['page_id']).specific
+
+    def get_default_success_url(self):
+        return reverse("wagtailadmin_explore", args=[self.get_object().get_parent().id])
+
+    def get_success_message(self, locales):
+        return _("The page '{page_title}' was successfully submitted for translation into {locales}").format(
+            page_title=self.object.get_admin_display_title(),
+            locales=locales
+        )
+
+
+class SubmitSnippetTranslationView(SubmitTranslationView):
+
+    def get_title(self):
+        return _("Translate {model_name}").format(
+            model_name=capfirst(self.object._meta.verbose_name)
```

Comment:
Playing devils advocate here: should all model names be capped in every instance? I've found that this covers _most_ instances but there are sometimes outliers where the model name is purposely lowercased all the time (when displaying model names)

Reply:
Actually, yeah that makes sense. Probably only need to do this when we use the model name at the beginning of a sentence on its own.